sql-to-dbsp-compiler/lib/sqllib/src/timestamp.rs

```diff
@@ -836,6 +836,50 @@ impl Time {
         )
         .unwrap()
     }
+
+    pub const MAX: Time = Time {
+        nanoseconds: 86_399_000_000_000, /*23:59:59*/
+    };
+}
+
+impl Add<ShortInterval> for Time {
+    type Output = Self;
+
+    fn add(self, rhs: ShortInterval) -> Self::Output {
+        // 24:00:00 or 00:00:00
+        let nanos_in_day = (Time::MAX.nanoseconds() + BILLION) as i64;
+
+        let mut time = self.nanoseconds() as i64;
+        let int = rhs.nanoseconds() % nanos_in_day;
+
+        if time == 0 {
+            time = nanos_in_day;
+        }
+
+        let rem = (time + int) % nanos_in_day;
```

Comment:
on the contrary, the interval may be negative, please make sure that modulo does the expected thing.